module/Category/src/Controller/CategoryController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Category\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+use Doctrine\ORM\EntityManager;
+use Category\Repository\CategoryRepository;
+
+class CategoryController extends AbstractActionController
+{
+    private $entityManager;
+
+    public function __construct($serviceManager)
+    {
+       $this->entityManager = $serviceManager->get(EntityManager::class);
+    }
+
+    public function indexAction()
+    {
+        $categories = new CategoryRepository($this->entityManager);
+        $data = $categories->findAll();
+        return new ViewModel(['categories' => $data]);
+    }
+
+    public function addAction()
+    {
+        return new ViewModel();
+    }
+
+    public function addAjaxAction()
+    {
+        if($this->getRequest()->isPost()) {
+           $category = new CategoryRepository($this->entityManager);
+           // Fill in the form with POST data
+           $data = $this->params()->fromPost();
+           // var_dump($data);
+           $category->create($data);
+        }
+        die;
```

Comment:
You should not `die` inside a controller. Please see this URL about ways of returning different output (i.e. not HTML):
https://olegkrivtsov.github.io/using-zend-framework-3-book/html/en/Model_View_Controller/View_Rendering_Strategies.html